logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationInteractionAvailabilityUseCase.kt

```diff
@@ -0,0 +1,63 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationDetails
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.user.ConnectionState
+import com.wire.kalium.logic.functional.fold
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+/**
+ * Use case that check if self user is able to interact in conversation
+ * @param conversationId the id of the conversation where user checks his interaction availability
+ * @return an [IsInteractionAvailableResult] containing Success or Failure cases
+ */
+class ObserveConversationInteractionAvailabilityUseCase internal constructor(
+    private val conversationRepository: ConversationRepository
+) {
+    suspend operator fun invoke(conversationId: ConversationId): Flow<IsInteractionAvailableResult> {
+        return conversationRepository.observeConversationDetailsById(conversationId).map { eitherConversation ->
+            eitherConversation.fold({ failure -> IsInteractionAvailableResult.Failure(failure) }, { conversationDetails ->
+                val availability = when (conversationDetails) {
+                    is ConversationDetails.Connection -> InteractionAvailability.DISABLED
+                    is ConversationDetails.Group -> if (conversationDetails.isSelfUserMember) InteractionAvailability.ENABLED
+                    else InteractionAvailability.NOT_MEMBER
```

Comment:
This `else` here is really confusing!

It looks like it is the `else` of the `when` operator and not the `else` of the `if`.

```suggestion
                    is ConversationDetails.Group -> {
                        if (conversationDetails.isSelfUserMember) InteractionAvailability.ENABLED
                        else InteractionAvailability.NOT_MEMBER
                    }
```